### PR TITLE
Catch up embulk 0.10

### PR DESCRIPTION
**Pull request by ngochc**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,20 +12,51 @@ repositories {
...
+        exclude group: 'com.fasterxml.jackson.datatype', module: 'jackson-datatype-jdk8'
+        exclude group: 'javax.validation', module: 'validation-api'
+    }
+    compile 'org.embulk:embulk-util-retryhelper:0.8.1'
+    compile "org.embulk:embulk-util-json:0.1.0"
+
+    // embulk 10.31 has problem with null value
+    compile "org.embulk:embulk-guess-csv:0.10.36"
+    compile "org.embulk:embulk-parser-csv:0.10.36"
+    compile "org.embulk:embulk-input-file:$embulkVersion"
```
Comment (dmikurube) [inline build.gradle]: Do we need them in `main`, not `test`?

File: src/test/java/org/embulk/input/zendesk/TestZendeskInputPlugin.java [hunk 2]
```diff
@@ -428,8 +432,20 @@ private void setupTestGuessGenerateColumn(ConfigSource src, String fileName, Str
...
         ConfigDiff configDiff = zendeskInputPlugin.guess(src);
+        List<String> actualValues = new ArrayList<>();
         JsonNode columns = configDiff.get(JsonNode.class, "columns");
-        assertEquals(ZendeskTestHelper.getJsonFromFile(expectedSource), columns);
+        Iterator<JsonNode> it = columns.elements();
+        while (it.hasNext()) {
+            JsonNode actual = it.next();
+            actualValues.add(actual.toString());
+        }
+        System.out.println(actualValues);
```
Comment (hieudion) [inline src/test/java/org/embulk/input/zendesk/TestZendeskInputPlugin.java]: Pl remove it

File: src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java [hunk 3]
```diff
@@ -280,64 +298,143 @@ private JsonNode buildColumns(final PluginTask task)
...
+        ConfigDiff configDiff = CONFIG_MAPPER_FACTORY.newConfigDiff();
+        configDiff.setNested("parser", parserGuessed);
+        return configDiff;
+    }
+
+    private List<List<Object>> createSampleData(JsonNode jsonNode, String targetJsonName, List<String> unifiedFieldNames)
+    {
+        final List<List<Object>> samples = new ArrayList<>();
+        Iterator<JsonNode> records = ZendeskUtils.getListRecords(jsonNode, targetJsonName);
+        while (records.hasNext()) {
```
Comment (hieudion) [inline src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java:366]: For guess phase, we should limit the records from 5 to 10 records

File: build.gradle (hunk 1 above)
Comment (ngochc) [reply build.gradle]: It is redundant let me remove it. Thanks

File: src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java [hunk 4]
```diff
@@ -280,64 +298,143 @@ private JsonNode buildColumns(final PluginTask task)
...
+            samples.add(line);
+        }
+        return samples;
+    }
+
+    private List<String> unifiedFieldNames(JsonNode jsonNode, String targetJsonName)
+    {
+        List<String> columnNames = new ArrayList<>();
+        Iterator<JsonNode> records = ZendeskUtils.getListRecords(jsonNode, targetJsonName);
+        while (records.hasNext()) {
```
Comment (hieudion) [inline src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java:391]: I think for unified field name, we don't need to go through all the records, just one or a few ones are enough

**Review by hieudion — CHANGES_REQUESTED** [review]

File: src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java (hunk 3 above)
Comment (ngochc) [reply src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java:366]: As our discussion, I test it on development env and it work well with page (~100 records)

File: src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java (hunk 4 above)
Comment (ngochc) [reply src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java:391]: As our discussion, I test it on development env and it work well with page (~100 records)

**Review by hieudion — APPROVED** [review]
LGTM

File: gradle/dependency-locks/embulkPluginRuntime.lockfile [hunk 5]
```diff
@@ -1,7 +1,24 @@
...
+com.fasterxml.jackson.core:jackson-annotations:2.6.7
+com.fasterxml.jackson.core:jackson-core:2.6.7
+com.fasterxml.jackson.core:jackson-databind:2.6.7
+com.fasterxml.jackson.datatype:jackson-datatype-guava:2.6.7
+com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7
+com.google.guava:guava:18.0
+com.ibm.icu:icu4j:54.1.1
 commons-codec:commons-codec:1.10
+commons-io:commons-io:2.7
 commons-logging:commons-logging:1.2
```
Comment (dmikurube) [inline gradle/dependency-locks/embulkPluginRuntime.lockfile]: We may want to find where this `commons-logging` come from, and replace it with `jcl-over-slf4j:1.7.12`. See https://github.com/embulk/embulk-output-gcs/pull/44 for reference.

File: gradle/dependency-locks/embulkPluginRuntime.lockfile [hunk 6]
```diff
@@ -1,7 +1,24 @@
...
 org.apache.httpcomponents:httpclient:4.5.6
 org.apache.httpcomponents:httpcore:4.4.10
+org.embulk:embulk-util-config:0.3.1
+org.embulk:embulk-util-file:0.1.3
+org.embulk:embulk-util-guess:0.2.0
+org.embulk:embulk-util-json:0.1.0
+org.embulk:embulk-util-retryhelper:0.8.1
+org.embulk:embulk-util-text:0.1.1
+org.json:json:20200518
+org.msgpack:msgpack-core:0.8.11
```
Comment (dmikurube) [inline gradle/dependency-locks/embulkPluginRuntime.lockfile]: `org.msgpack.msgpack-core` should not be included here. Please try finding where it comes from, and `exclude` it.

**Review by dmikurube — APPROVED** [review]
Sorry to be late. Approved it not to unblock you, but left two comments. Do the merge after addressing these two comments.

**Pull request merged**